les/api.go

```diff
@@ -52,6 +55,70 @@ func NewPrivateLightServerAPI(server *LesServer) *PrivateLightServerAPI {
 	}
 }
 
+//GatewayFee returns the current gateway fee of this light server
+func (api *PrivateLightServerAPI) GatewayFee() (gf *big.Int, err error) {
+	return api.server.handler.gatewayFee, nil
+}
+
+//SetGatewayFee allows this light server node to set a gateway fee
+func (api *PrivateLightServerAPI) SetGatewayFee(gf *big.Int) error {
+	if gf.Cmp(big.NewInt(0)) == -1 {
+		return errors.New("Invalid Gateway Fee Value")
+	}
+	api.server.handler.gatewayFee = gf
+	if err := api.BroadcastGatewayFeeUpdate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetEtherbase sets the etherbase of the miner. Not sure if there should be one here or not because theres already one in the eth api
+func (api *PrivateLightServerAPI) SetEtherbase(etherbase common.Address) common.Address { //could do the casting thing again for test. Also, sending push notif here. Generally assume not changing etherbase. but if it can, need more checks on cache get
+	api.server.handler.etherbase = etherbase
+	if etherbase != common.ZeroAddress {
+		api.BroadcastGatewayFeeUpdate()
+	}
+	return api.server.handler.etherbase
+}
+
+//get the unique node id of current node
+func (api *PrivateLightServerAPI) NodeID() string {
+	return api.server.handler.nodeID
+}
+
+//This sends messages to light client peers whenever this light server updates gateway fee.
+func (api *PrivateLightServerAPI) BroadcastGatewayFeeUpdate() error {
```

Comment:
Id say this should at least be renamed to `BroadcastGatewayFee` (it may be used even in absence of an update). Additionally, it should be moved out of the API and into a more core piece of the logic like we discussed.